src/main/java/com/techsophy/tsf/services/gateway/config/AwgmentOAuthRequestResolver.java

```diff
@@ -0,0 +1,66 @@
+package com.techsophy.tsf.services.gateway.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.server.DefaultServerOAuth2AuthorizationRequestResolver;
+import org.springframework.security.oauth2.client.web.server.ServerOAuth2AuthorizationRequestResolver;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+import static com.techsophy.tsf.services.gateway.constants.GatewayConstants.*;
+
+@Component
+public class AwgmentOAuthRequestResolver implements ServerOAuth2AuthorizationRequestResolver
+{
+    private final ServerOAuth2AuthorizationRequestResolver defaultResolver;
+    private final String keycloakIssuerUri;
+    private final String defaultRealmName;
+
+    public AwgmentOAuthRequestResolver(ReactiveClientRegistrationRepository repo,
+            @Value("${keycloak.issuer-uri}") String keycloakIssuerUri,
+            @Value("${keycloak.realm.name}")String realmName)
+    {
+        this.defaultResolver = new DefaultServerOAuth2AuthorizationRequestResolver(repo);
+        this.defaultRealmName = realmName;
+        this.keycloakIssuerUri=keycloakIssuerUri;
+    }
+
+    @Override
+    public Mono<OAuth2AuthorizationRequest> resolve(ServerWebExchange exchange)
+    {
+        String header = exchange.getRequest().getHeaders().getFirst(X_TENANT);
```

Comment:
both functions are duplicate code, please create refactor to reuse.

Reply:
yes refactoring code